components/compliance-service/inspec-agent/resolver/resolver.go

```diff
@@ -582,6 +582,12 @@ func (r *Resolver) resolveDynamicJob(ctx context.Context, job *jobs.Job) ([]*typ
 		}
 		jobs = append(jobs, agentJobs...)
 	}
+
+	// Keeping the child - parent job reference alive in the resolved jobs array
+	for _, j := range jobs {
+		j.ParentJobID = job.Id
+	}
```

Comment:
We can technically remove this again now that resolve-job happens as part of the workflow, but we figured that it is nice to keep the parent ID around for logging and debugging.